jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -312,6 +315,19 @@ private Response call(String httpMethod, URL url, Request request, HttpTransport
         httpTransport
             .createRequestFactory()
             .buildRequest(httpMethod, new GenericUrl(url), request.getHttpContent())
+            .setIOExceptionHandler(new HttpBackOffIOExceptionHandler(new ExponentialBackOff()) {
+              @Override
+              public boolean handleIOException(final HttpRequest request, final boolean supportsRetry) throws IOException {
+                final boolean result = super.handleIOException(request, supportsRetry);
+                final String reqString = request.getRequestMethod() + " " + request.getUrl();
+                if (result) { // google-http-client does not log that properly so let's compensate it
+                  logger.accept(LogEvent.warn(reqString + " failed and will be retried"));
+                } else {
+                  logger.accept(LogEvent.warn(reqString + " failed and will NOT be retried"));
```

Comment:
I think the `else` branch can be debug (or removed). 
```suggestion
                  logger.accept(LogEvent.warn(reqString + " failed due to IOException and will be retried"));
                } else {
                  logger.accept(LogEvent.debug(reqString + " failed due to IOException and will NOT be retried"));
```

Reply:
FTR: https://github.com/GoogleContainerTools/jib/pull/3351#discussion_r683571069